### PR TITLE
backend changes for deployment

### PR DESCRIPTION
**Pull request by nataliemona**
### Please note breaking changes!!!!

* Move your backend/core/.env file to backend/.env
* requirements-dev.txt is now gone
* In the backend folder. Instead of a single requirements.in file, there is a requirements folder that includes "base" and "dev" requirements. This splits the dependencies up between dependencies needed for development and testing and dependencies needed to run the application

* To install dev dependencies: `pip-sync requirements/dev.txt` or `pip install -r requirements/dev.txt`
* Add your dependencies to the `.in` files and update the `.txt` files with `pip-compile requirements/dev.in` or `pip-compile requirements/base.in` (base requirements command is automatically run when you run your docker container so really you should only ever have to run `pip-compile requirements/dev.in`)

### CODE REVIEW

File: backend/uwsgi.ini [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[uwsgi]
```
Comment (nataliemona) [inline backend/uwsgi.ini:1]: This is a configuration file for the uwsgi server. https://uwsgi-docs.readthedocs.io/en/latest/WSGIquickstart.html and https://docs.djangoproject.com/en/3.1/howto/deployment/wsgi/uwsgi/ if you are curious

File: docker-compose.yml [hunk 2]
```diff
@@ -13,19 +13,27 @@ services:
...
-    build: ./backend
+    env_file:
+      - ./backend/.env
+    environment:
+      - MODE=dev
+      - DB_HOST=pgdb
+      - DB_PORT=5432
+    build: 
+      context: ./backend
+      target: base
```
Comment (nataliemona) [inline docker-compose.yml:26]: `target: base` means that when you docker-compose up only the part of the Dockerfile labeled "base" will run (i.e. not the server)

File: backend/requirements/base.in [hunk 3]
```diff
@@ -4,10 +4,5 @@ django_elasticsearch_dsl_drf==0.21
 psycopg2-binary>=2.8
 python-dotenv
 djangorestframework
-markdown
```
Comment (nataliemona) [inline backend/requirements/base.in:7]: looks like this is not being used

**Review by BlanchardMR — COMMENTED** [review]
Just wondering if there is a good reason to divide the dev and runtime requirements for this project. Will this lead to more or less confusion? Does it increase performance by not loading so many libraries?

**Review by BlanchardMR — APPROVED** [review]
build and works for me

**Comment by nataliemona** [comment]
Something is messed up with this PR. It's not updating with my latest merge commit.

**Pull request closed without merging**